cpp/oneapi/dal/table/backend/convert.cpp

```diff
@@ -227,33 +227,50 @@ sycl::event convert_vector_device2host(sycl::queue& q,
     // To perform conversion, we gather data from device to host in temporary
     // contigious array and then run host conversion function
 
-    const std::int64_t element_size_in_bytes = dal::detail::get_data_type_size(src_type);
+    const std::int64_t dst_element_size_in_bytes = dal::detail::get_data_type_size(dst_type);
+    const std::int64_t dst_size_in_bytes =
+        dal::detail::check_mul_overflow(dst_element_size_in_bytes, element_count);
+    const std::int64_t dst_stride_in_bytes =
+        dal::detail::check_mul_overflow(dst_element_size_in_bytes, dst_stride);
+
+    const std::int64_t src_element_size_in_bytes = dal::detail::get_data_type_size(src_type);
     const std::int64_t src_size_in_bytes =
-        dal::detail::check_mul_overflow(element_size_in_bytes, element_count);
+        dal::detail::check_mul_overflow(src_element_size_in_bytes, element_count);
     const std::int64_t src_stride_in_bytes =
-        dal::detail::check_mul_overflow(element_size_in_bytes, src_stride);
-
-    const auto tmp_host_unique = make_unique_usm_host(q, src_size_in_bytes);
-
-    auto gather_event = gather_device2host(q,
-                                           tmp_host_unique.get(),
-                                           src_device,
-                                           element_count,
-                                           src_stride_in_bytes,
-                                           element_size_in_bytes,
-                                           deps);
-    gather_event.wait_and_throw();
-
-    convert_vector(dal::detail::default_host_policy{},
-                   tmp_host_unique.get(),
-                   dst_host,
-                   src_type,
-                   dst_type,
-                   1L,
-                   dst_stride,
-                   element_count);
-
-    return sycl::event{};
+        dal::detail::check_mul_overflow(src_element_size_in_bytes, src_stride);
+    if (src_element_size_in_bytes == dst_element_size_in_bytes &&
```

Comment:
I guess there should be something like this. Because `int32` and `float` will have `src_element_size_in_bytes` == `dst_element_size_in_bytes`.

```suggestion
    if (src_type == dst_type &&
```